README.md

```diff
@@ -169,6 +173,35 @@ User.reindex
 Depending on how much data you have, reindexing may take a while but you only need to do it once.
 *ActiveStash will automatically index (and delete) data as it records are created, updated and deleted.*
 
+## Current limitations
+
+Presently, ActiveStash provides no means to update the schema of a CipherStash collection. Therefore if you need to make any changes to the Collection schema itself (by using the `stash_index` or `stash_match_all` helpers) you must drop your collection and recreate it.
+
+If your indexed model is called `User` for example, you should run the following commands:
+
+```sh
+$ rails active_stash:drop[User]
+$ rails active_stash:create
+$ rails active_stash:reindex[User]
+```
+
+Support for zero-downtime Collection schema changes and reindexing is being actively worked on and will be available soon.
+
+### When to Reindex Your Collection
+
+These are the rules for when you *must* re-index your collection:
+
+1. You have imported, deleted or updated data in the table that backs your ActiveStash model via some external mechanism, OR
```

Comment:
Based on a quick glance at the `Search#reindex` method, I don't think that *deleted* data actually gets accounted for during the reindex process.  That is, if the dev does a `DELETE FROM users;` and then runs `rake active_stash:reindex[User]`, the `users` collection will still have data in it for all the users that got deleted.

Might be worth double-checking with @coderdan as to whether I'm reading that right before merging, just in case.

Reply:
I've tried to raise @coderdan's attention on this. It sounds like something we should address fairly urgently.

Reply:
So if a user does delete all from users they really should drop and recreate the collection (under the current model anyway).

Reply:
Callbacks don't fire on delete all

Reply:
@coderdan the issue is that a reindex doesn't take into account deleted records. The deleted records will still be findable in CipherStash

Reply:
The scenario we're taking about here is that someone deletes records _outside_ of their app (directly in the DB, for example). You'd _think_ that a reindex would fix that but it won't. It will not tell CipherStash to delete old records. Only the records presently in the database will have their indexes refreshed.

Reply:
Notion card: https://www.notion.so/cipherstash/ActiveStash-reindex-should-drop-deleted-records-from-the-index-f90a8e37649844f99b5e37020c7fb438